### ISSUE TITLE
[Analyzer Proposal]: Warn unnecessary null-forgiving operators

### ISSUE THREAD

**Issue by alexandrehtrb** [dotnet/roslyn#77060]
**Brief description:**

Nullable type checking in .NET is a great feature that was added a few years ago.
However, people sometimes overuse null-forgiving operators, applying them when they are not needed.
Overuse of null-forgiving operators may hide problems after refactors, when a non-nullable variable becomes nullable.

Consider the code below:

```cs
public class FootballPlayer
{
    public string Name { get; }
    public FootballTeam Team { get; }
}

public class FootballTeam
{
    public string Name { get; }
    public List<FootballPlayer> Players { get; }
}

...

string playerTeamName = player.Team!.Name!;
```

If the player's team is changed to nullable, a NullReferenceException may happen when accessing `player.Team!.Name!`, but this won't be noticed (a sneaky bug).

**Languages applicable:**

C#, maybe F# and VB.NET too?

**Code example that the analyzer should report:**

```cs
string myName = "John McClane";

string anotherName = myName!;
// info or warning: unnecessary null-forgiving operator.
```

**Additional information:**

Any more additional information you would like to add.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by RikkiGibson** [comment]
Thanks for opening this issue. It looks like this is a duplicate, so will close out. Please feel free to continue discussion in the other issue.
